pom.xml

```diff
@@ -1440,5 +1376,11 @@
                 <spark.archive.download.skip>true</spark.archive.download.skip>
             </properties>
         </profile>
+        <profile>
+            <id>kyuubi-shaded</id>
+            <modules>
+                <module>kyuubi-shaded</module>
```

Comment:
why it is a profile?

Reply:
After some investigation with @turboFei, we found that maven has some limitation of shade module reference, when run `mvn package`, the downstream module can't see shaded classes of upstream module which in same maven project.

There is a solution https://stackoverflow.com/a/50274253 but introduced another issue, we must run `mvn clean xxx` everytime, w/o `mvn clean` will cause compile failed.

Maybe that's why Apache Flink and Apache Hadoop choose shade deps in separated module, and Apache Iceberg which use Gradle doesn't suffer this issue.

Reply:
So this do related to the releasing process and a doc is also needed

Reply:
So this is related to the releasing process, and a documentation is also needed

Reply:
is it ok to add some notes into building doc?

Reply:
https://github.com/yaooqinn/kyuubi/pull/488/files?file-filters%5B%5D=.md&file-filters%5B%5D=.scala&file-filters%5B%5D=.xml#diff-434e619c9343e148d9ef7e85b044633eb86f72c6f1137eac287a97708f18c86cR11